File: src/main/java/com/chrisdempewolf/responses/pin/Counts.java
package com.chrisdempewolf.responses.pin;

public class Counts {
    private final Integer likes;
    private final Integer comments;
    private final Integer repins;

    public Counts(Integer likes, Integer comments, Integer repins) {
        this.likes = likes;
        this.comments = comments;
        this.repins = repins;
    }

    public Integer getLikes() {
        return likes;
    }

    public Integer getComments() {
        return comments;
    }

    public Integer getRepins() {
        return repins;
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (o == null || getClass() != o.getClass()) return false;

        Counts counts = (Counts) o;

        if (likes != null ? !likes.equals(counts.likes) : counts.likes != null) return false;
        if (comments != null ? !comments.equals(counts.comments) : counts.comments != null) return false;
        return !(repins != null ? !repins.equals(counts.repins) : counts.repins != null);

    }

    @Override
    public int hashCode() {
        int result = likes != null ? likes.hashCode() : 0;
        result = 31 * result + (comments != null ? comments.hashCode() : 0);
        result = 31 * result + (repins != null ? repins.hashCode() : 0);
        return result;
    }

    @Override
    public String toString() {
        return "Counts{" +
                "likes=" + likes +
                ", comments=" + comments +
                ", repins=" + repins +
                '}';
    }
}

File: src/main/java/com/chrisdempewolf/pinterest/responses/pin/Counts.kt
package com.chrisdempewolf.pinterest.responses.pin

class Counts(val likes: Int?, val comments: Int?, val repins: Int?) {

    override fun equals(o: Any?): Boolean {
        if (this === o) return true
        if (o == null || javaClass != o.javaClass) return false

        val counts = o as Counts?

        if (if (likes != null) likes != counts!!.likes else counts!!.likes != null) return false
        if (if (comments != null) comments != counts.comments else counts.comments != null) return false
        return !if (repins != null) repins != counts.repins else counts.repins != null

    }

    override fun hashCode(): Int {
        var result = if (likes != null) likes.hashCode() else 0
        result = 31 * result + if (comments != null) comments.hashCode() else 0
        result = 31 * result + if (repins != null) repins.hashCode() else 0
        return result
    }

    override fun toString(): String {
        return "Counts{" +
                "likes=" + likes +
                ", comments=" + comments +
                ", repins=" + repins +
                '}'
    }
}
